File: Benchmarking/src/gen129.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    s0(std::vector<std::vector<double>> v0): p0(v0) { }
  };
  __attribute__((noinline))
  double f0(const std::vector<s0> &v0, const double &v1) {
    const s0 v2 = v0[0];
    const s0 v14 = v0[0];
    const std::vector<std::vector<double>> v12 = v14.p0;
    const std::vector<double> v24 = v12[0];
    const std::vector<double> v19 = v12[0];
    const std::vector<std::vector<double>> v26 = v2.p0;
    const std::vector<double> v22 = v26[0];
    const std::vector<double> v200 = v12[0];
    std::vector<double> v108 = v22;
    const std::vector<std::vector<double>> v257 { v200, v108, v19 };
    std::vector<std::vector<double>> v148 = v257;
    std::vector<std::vector<double>> v58 = v148;
    std::vector<std::vector<double>> v204 = v58;
    const std::vector<double> v265 = v204[2];
    const double v216 = v265[0];
    v204[0] = v108;
    v204[0] = v24;
    return v216;
  }
  int main() {
    std::vector<s0> v0({ { { { 0.0 } } } });
    double v1(1.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }

File: Benchmarking/src/gen129.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
  }
  @inline(never)
  func f0(_ v0: [s0], _ v1: Double) -> Double {
    let v2: s0 = v0[0]
    let v14: s0 = v0[0]
    let v12: [[Double]] = v14.p0
    let v24: [Double] = v12[0]
    let v19: [Double] = v12[0]
    let v26: [[Double]] = v2.p0
    let v22: [Double] = v26[0]
    let v200: [Double] = v12[0]
    var v108: [Double] = v22
    let v257: [[Double]] = [v200, v108, v19]
    var v148: [[Double]] = v257
    var v58: [[Double]] = v148
    var v204: [[Double]] = v58
    let v265: [Double] = v204[2]
    let v216: Double = v265[0]
    v204[0] = v108
    v204[0] = v24
    return v216
  }
  func benchmark() {
    let v0: [s0] = [s0(p0: [[0.0]])]
    let v1: Double = 1.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 1.0
    for _ in 1...1000 {
      result = f0(v0, v1)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()
